docs/resources/secondary_database.md

```diff
@@ -0,0 +1,82 @@
+---
+page_title: "snowflake_secondary_database Resource - terraform-provider-snowflake"
+subcategory: ""
+description: |-
+  A secondary database creates a replica of an existing primary database (i.e. a secondary database). For more information about database replication, see Introduction to database replication across multiple accounts https://docs.snowflake.com/en/user-guide/db-replication-intro.
+---
+
+# snowflake_secondary_database (Resource)
+
+A secondary database creates a replica of an existing primary database (i.e. a secondary database). For more information about database replication, see [Introduction to database replication across multiple accounts](https://docs.snowflake.com/en/user-guide/db-replication-intro).
+
+## Example Usage
+
+```terraform
+resource "snowflake_secondary_database" "test" {
+  name          = "database_name"
+  as_replica_of = "organization_name.account_name.primary_database_name"
+  is_transient  = false
+
+  data_retention_time_in_days {
+    value = 10
+  }
+
+  max_data_extension_time_in_days {
+    value = 20
+  }
+
+  external_volume       = "external_volume_name"
+  catalog               = "catalog_name"
+  default_ddl_collation = "en_US"
+  log_level             = "OFF"
+  trace_level           = "OFF"
+  comment               = "A secondary database"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `as_replica_of` (String) A fully qualified path to a database to create a replica from. A fully qualified path follows the format of `"<organization_name>"."<account_name>"."<database_name>"`.
```

Comment:
Yes, but I changed to the documentation one `<provider_account>.<share_name>`, because we should be able to handle both identifiers. I'll modify one integration and acceptance test to show it's working.

Reply:
On second thought, we can easily support both formats on SDK level, but it's hard to support both ways in Terraform because in Read we have to assume one format. For now, I'll revert the acceptance test, because it's failing for account locator and maybe we should enforce the new format (?).

Reply:
I think that it would be best to stick with the SF recommended way. This is a change against the old behavior but it is a new resource after all. However, we should mention this change in the migration notes.